pennylane/tape/tape.py

```diff
@@ -394,22 +394,17 @@ def _process_queue(self):
 
         Also calls `_update()` which sets many attributes.
         """
-        self._prep = []
-        self._ops = []
-        self._measurements = []
-        list_order = {"_prep": 0, "_ops": 1, "_measurements": 2}
-        current_list = "_prep"
-
-        for obj, info in self._queue.items():
-
-            if "owner" not in info and getattr(obj, "_queue_category", None) is not None:
-                if list_order[obj._queue_category] > list_order[current_list]:
-                    current_list = obj._queue_category
-                elif list_order[obj._queue_category] < list_order[current_list]:
-                    raise ValueError(
-                        f"{obj._queue_category[1:]} operation {obj} must occur prior "
-                        f"to {current_list[1:]}. Please place earlier in the queue."
-                    )
-                getattr(self, obj._queue_category).append(obj)
-
+        self._prep, self._ops, self._measurements = process_queue(self)
         self._update()
+
+    def __getitem__(self, key):
+        try:
+            return QuantumScript.__getitem__(self, key)
+        except TypeError:
+            return AnnotatedQueue.__getitem__(self, key)
+
+    def __setitem__(self, key, val):
+        AnnotatedQueue.__setitem__(self, key, val)
```

Comment:
Are we sure we want to allow this? Tapes are meant to be immutable.

Reply:
this (and other dunder changes) are here because we use QuantumTape as if it's a queue, which is the whole problem we're trying to solve. `AnnotatedQueue.append` will set a key to a certain value, but `QuantumTape` inherits `QuantumScript` first, and `AnnotatedQueue` second. The following call stack happens often and causes trouble without this:
```python
QueuingManager.append(obj)
QueuingManager._active_contexts[-1].append(obj)
QuantumTape.append(obj)  # note that QuantumTape inherits AnnotatedQueue, this is a context
QuantumTape[obj] = {}
```

To clarify your concern: _scripts_ are meant to be immutable, _queues_ are not. Unfortunately, `QuantumTape` is both!

Reply:
Thank you!

> scripts are meant to be immutable, queues are not. Unfortunately, QuantumTape is both!

I think this is an important shift in the characteristic of a main abstraction and would be worth bringing up at a team discussion :+1: